src/js/stores/MesosStateStore.js

```diff
@@ -179,10 +187,21 @@ class MesosStateStore extends GetSetBaseStore {
     let services = this.get('lastMesosState').frameworks || [];
     let memberTasks = {};
 
-    services.forEach(function (service) {
+    const schedulerTaskIds = services.reduce((memo, service) => {
+      const serviceName = service.name.split('.').reverse().join('/');
+      const schedulerTask = this.getSchedulerTaskFromServiceName(serviceName);
+
+      if (schedulerTask) {
+        memo.push(schedulerTask.id);
+      }
+
+      return memo;
+    }, []);
+
+    services.forEach((service) => {
       service.tasks.forEach(function (task) {
         if (task.slave_id === nodeID) {
-          memberTasks[task.id] = task;
+          memberTasks[task.id] = isSchedulerTask(task, schedulerTaskIds);
```

Comment:
wonder if there is way to flag the scheduler tasks in one go instead of iterating same list over and over again